Frameworks/CoreText/CTRun.mm

```diff
@@ -88,167 +55,149 @@ DWRITE_GLYPH_RUN __GetGlyphRunForDrawingInRange(const DWRITE_GLYPH_RUN& run, con
 @Status Interoperable
 */
 CFIndex CTRunGetGlyphCount(CTRunRef run) {
-    return run ? static_cast<_CTRun*>(run)->_dwriteGlyphRun.glyphCount : 0;
+    RETURN_RESULT_IF_NULL(run, 0);
+    return run->_dwriteGlyphRun.glyphCount;
 }
 
 /**
 @Status Interoperable
 */
 CFDictionaryRef CTRunGetAttributes(CTRunRef run) {
-    return run ? static_cast<CFDictionaryRef>(static_cast<_CTRun*>(run)->_attributes.get()) : nil;
+    RETURN_NULL_IF(!run);
```

Comment:
you sure you wouldn't prefer a nice clean ternary here? 😉 
`return run ? run->_attributes : nil;`